Review for Portuguese translation of `maintenance_collaboration.pt.Rmd` (#722)
ropensci/dev_guide: maintenance_collaboration.pt.Rmd

Esses links fornecem um modelo para uma declaração de ciclo de vida simples.
Os arquivos `CONTRIBUTING.md` também podem descrever como você reconhece as contribuições (consulte [esta seção](#attributions)).

Recomendamos que você envie comentários que não sejam um relatório de bug ou uma solicitação de *feature* para o [Fórum da rOpenSci](https://discuss.ropensci.org/) após certificar-se de que você verá essas perguntas no fórum. Os usuários podem usar o fórum para fazer perguntas sobre o uso e relatar seus casos de uso, e você pode se inscrever em categorias e tags individuais para receber notificações sobre o seu pacote.  Sinta-se à vontade para mencionar isso nos documentos do seu pacote e/ou nas diretrizes de contribuição/modelo de *issue*. Oriente os seus usuários a marcar as mensagens com o nome do pacote.

Quando uma pull request estiver mais perto de ser mesclada, você poderá usar [um *GitHub Actions PR workflow* para estilizar o código com o pacote `styler`](https://github.com/r-lib/actions/blob/master/examples/pr-commands.yaml).

### Gerenciamento de *issues* {#issue-management}

> Ao usar os recursos do GitHub em torno dos *issues*, você pode ajudar os possíveis colaboradores a encontrá-los e tornar público o seu roteiro.

Reviewer: ```suggestion
Ao usar os recursos do GitHub em torno das *issues*, você pode ajudar os possíveis colaboradores a encontrá-los e tornar público o seu roteiro.
```

Issue eu sempre entendi como palavra no feminino, mas agora tô em dúvida. Marcando aqui pra discussão.
Olhei na documentação oficial e eles traduziram como "o problema" 🤡 , mas ninguém chama assim.
https://docs.github.com/pt/issues/tracking-your-work-with-issues/creating-an-issue

Author: De fato, é comum termos da programação ficarem meio esquisitos quando traduzidos direto para o portugues.

Mas de qualquer modo, eu entendo que o termo "problema" é adequado quando usado dentro do contexto. Em outras palavras, o contexto sobre o que estamos discutindo ajuda a compreender o significado do termo "problema". Como estamos discutindo nesse capítulo sobre a colaboração na manutenção de pacotes, fica mais fácil de entender que "problema" se refere às issues do GitHub onde uma parte da discussão e da colaboração acontece.

#### Modelos de *issues* {#issue-templates}

> Você pode usar um ou vários [modelo(s) de *issue(s)*](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-templates=) para ajudar os usuários a preencherem os relatórios de bugs ou as solicitações de *feature*.

Reviewer: ```suggestion
Você pode usar um ou vários [modelo(s) de *issue(s)*](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-templates=) para ajudar as pessoas usuárias a preencherem os relatórios de bugs ou as solicitações de *feature*.
```

> Quando há vários modelos de *issues*, os usuários que clicam em abrir uma nova *issue* veem um menu que orienta as suas escolhas.

Reviewer: ```suggestion
Quando há vários modelos de *issues*, as pessoas usuárias que clicam em abrir uma nova *issue* veem um menu que orienta as suas escolhas.
```

Você pode até [configurar uma das opções](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#configuring-the-template-chooser=) para apontar para algum lugar fora do seu repositório (por exemplo, um fórum de discussão).

Consulte a [Documentação do GitHub](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository).

#### Rotulagem de *issues* {#issue-labelling}

> Você pode usar rótulos como "help wanted" (procura-se ajuda) e "good first issue" (bom primeiro problema) para ajudar colaboradores em potencial, inclusive novatos, a encontrar o seu repositório. [Consulte o artigo do GitHub](https://help.github.com/articles/helping-new-contributors-find-your-project-with-labels/). Você também pode usar o rótulo "Beginner" (Iniciante). Você pode ver [exemplos de problemas para iniciantes em todos os repositórios da rOpenSci](https://github.com/search?q=user%3Aropensci+user%3Aropenscilabs+label%3ABeginner+state%3Aopen&type=Issues).

Reviewer: ```suggestion
Você pode usar rótulos como "help wanted" (procura-se ajuda) e "good first issue" (boa primeira *issue*) para ajudar pessoas colaboradoras em potencial, inclusive iniciantes, a encontrar o seu repositório. [Consulte o artigo do GitHub](https://help.github.com/articles/helping-new-contributors-find-your-project-with-labels/). Você também pode usar o rótulo "Beginner" (Iniciante). Você pode ver [exemplos de problemas para iniciantes em todos os repositórios da rOpenSci](https://github.com/search?q=user%3Aropensci+user%3Aropenscilabs+label%3ABeginner+state%3Aopen&type=Issues).
```

#### Fixando *issues* {#pinning-issues}

Você pode [fixar até 3 *issues* por repositório](https://docs.github.com/en/articles/pinning-an-issue-to-your-repository) que aparecerão na parte superior do seu rastreador de *issues* como cartões de *issues* bonitos.
Isso pode ajudar a divulgar quais são suas prioridades.

#### Marcos {#milestones}

> Você pode [criar marcos](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones) e atribuir problemas a eles, o que ajuda outras pessoas a verem o que você planeja para a próxima versão do seu pacote, por exemplo.

Reviewer: ```suggestion
Você pode [criar marcos](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones) e atribuir *issues* a eles, o que ajuda outras pessoas a verem o que você planeja para a próxima versão do seu pacote, por exemplo.
```

> ### Comunicação com os usuários {#communication-with-users}

Reviewer: ```suggestion
### Comunicação com as pessoas usuárias {#communication-with-users}
```

> Você pode indicar aos usuários o fórum da rOpenSci se você monitorar ou ativar as [Discussões no GitHub](https://docs.github.com/en/discussions) para o seu repositório de pacotes.

Reviewer: ```suggestion
Você pode indicar para as pessoas usuárias o fórum da rOpenSci se você monitorar ou ativar as [Discussões no GitHub](https://docs.github.com/en/discussions) para o repositório do seu pacote.
```

Cada discussão do GitHub pode ser convertida em um *issue*, se necessário (e vice-versa, os *issues* podem ser convertidos em discussões).

## Trabalhando com colaboradores {#workingcollaborators}

Em primeiro lugar: mantenha contato com o seu repositório do GitHub!

> - Não se esqueça de [**observar o seu repositório do GitHub**](https://docs.github.com/en/github/managing-subscriptions-and-notifications-on-github/managing-subscriptions-for-activity-on-github) para ser notificado sobre *issues* ou *pull requests* (como alternativa, se você trabalha em rajadas, talvez adicione as informações ao guia de contribuição).

Reviewer: ```suggestion
- Não se esqueça de [**observar o seu repositório do GitHub**](https://docs.github.com/en/github/managing-subscriptions-and-notifications-on-github/managing-subscriptions-for-activity-on-github) para receber notificações sobre *issues* ou *pull requests* (como alternativa, se você trabalha de forma concentrada em certas épocas, talvez adicione essas informações ao guia de contribuição).
```

Tentei explicar melhor o que seria trabalhar em rajadas

- não se esqueça de enviar as atualizações que você tem localmente.

> - desative os testes com falha se você não puder corrigí-los, pois eles criam ruído nos PRs que podem confundir novos colaboradores.

Reviewer: ```suggestion
- desative os testes com falha se você não puder corrigí-los, pois eles criam ruído nos PRs que podem confundir colaboradores iniciantes.
```

> ### Integração de colaboradores {#onboarding-collaborators}

Reviewer: ```suggestion
### Integração de pessoas colaboradoras {#onboarding-collaborators}
```

> Não existe uma regra geral da rOpenSci sobre como você deve integrar colaboradores. Você deve aumentar os direitos deles sobre o repositório à medida que ganhar confiança e, sem dúvida, deve reconhecer as contribuições deles (consulte [esta seção](#attributions)).

Reviewer: ```suggestion
Não existe uma regra geral da rOpenSci sobre como você deve integrar pessoas colaboradoras. Você deve aumentar os direitos delas sobre o repositório à medida que ganhar confiança e, sem dúvida, deve reconhecer as contribuições que fizerem (consulte [esta seção](#attributions)).
```

> Você pode pedir a um novo colaborador que crie PRs (consulte a seção a seguir para avaliar um PR localmente, ou seja, além das verificações de CI) para dev/main e avalie-os antes de mesclá-los e, depois de algum tempo, deixe-os enviar para o main, embora você possa querer manter um sistema de revisões de PRs... mesmo para si mesmo quando tiver colegas de equipe!

Reviewer: ```suggestion
Você pode pedir para uma nova colaboradora que crie PRs (consulte a seção a seguir para avaliar um PR localmente, ou seja, além das verificações de CI) para dev/main e avalie-os antes de fazer *merge* e, depois de algum tempo, deixe-as enviar para a main, embora você possa querer manter um sistema de revisões de PRs... mesmo para si mesmo quando tiver colegas de equipe!
```

> Um modelo possível para a integração de colaboradores é fornecido por Jim Hester em [seu `lintr` repo](https://github.com/jimhester/lintr/issues/318).

Reviewer: ```suggestion
Um modelo possível para a integração de pessoas colaboradoras é fornecido por Jim Hester em [seu `lintr` repo](https://github.com/jimhester/lintr/issues/318).
```

> Se o seu problema for o recrutamento de colaboradores, você pode publicar uma chamada aberta como a de Jim Hester [no Twitter](https://twitter.com/jimhester_/status/997109466674819074), ou no [GitHub](https://github.com/jimhester/lintr/issues/318) e, como autor de um pacote da rOpenSci, você pode pedir ajuda no slack da rOpenSci e pedir à equipe da rOpenSci por ideias para recrutar novos colaboradores.

Reviewer: ```suggestion
Se o seu problema for o recrutamento de pessoas colaboradoras, você pode publicar uma chamada aberta como a de Jim Hester [no Twitter](https://twitter.com/jimhester_/status/997109466674819074), ou no [GitHub](https://github.com/jimhester/lintr/issues/318) e, como autor(a) de um pacote da rOpenSci, você pode pedir ajuda no slack da rOpenSci e pedir à equipe da rOpenSci por ideias para recrutar novas pessoas colaboradoras.
```

### Trabalhando com colaboradores (incluindo você) {#git-workflow}

[*Branchs* (ou ramificações)](https://happygitwithr.com/git-branches.html) são baratas. Use-as extensivamente ao desenvolver recursos, testar novas ideias e corrigir problemas.

> Uma das ramificações é a ramificação `main` (que é a ramificação padrão/principal), na qual, se você seguir [desenvolvimento baseado no tronco](https://www.atlassian.com/continuous-delivery/continuous-integration/trunk-based-development) você "mescla atualizações pequenas e frequentes".

Reviewer: ```suggestion
Uma das *branches* é a *branch* `main` (que é a ramificação padrão/principal), na qual, se você seguir [desenvolvimento baseado no tronco](https://www.atlassian.com/continuous-delivery/continuous-integration/trunk-based-development) você "faz *merge* de atualizações pequenas e frequentes".
```

Veja também as documentações do [Fluxo do GitHub](https://docs.github.com/en/get-started/quickstart/github-flow) e do [fluxo do GitLab](https://docs.gitlab.com/ee/topics/gitlab_flow.html).
Talvez você também queira incrementar frequentemente os números da versão de seu pacote (em `DESCRIPTION`).

> Um aspecto específico do trabalho com colaboradores é a revisão de solicitações de *pull*, com algumas orientações úteis em:

Reviewer: ```suggestion
Um aspecto específico do trabalho com colaboradores é a revisão de *Pull requests*, com algumas orientações úteis em:
```

- [Documentação do GitHub sobre revisões de PR](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/about-pull-request-reviews).

Você pode querer mexer nas configurações do seu repositório do GitHub para, por exemplo, [exigir revisões de *pull request* antes de fazer o *merge*](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches#require-pull-request-reviews-before-merging=).
Consulte também os documentos do GitHub sobre ["proprietários de código"](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners).

Para fazer e revisar *pull requests*, recomendamos que você [que você explore essas funções](https://usethis.r-lib.org/articles/pr-functions.html).

Para que você configure o "git remote", consulte [Happy Git and GitHub for the useR](https://happygitwithr.com/common-remote-setups.html).
Veja também a seção [Useful Git patterns for real life](https://happygitwithr.com/workflows-intro.html) no mesmo livro.

> ### Seja generoso com as atribuições {#attributions}

Reviewer: ```suggestion
### Seja generoso(a) com as atribuições {#attributions}
```

> Se alguém contribuir para o seu repositório, considere adicioná-lo em `DESCRIPTION`, como contribuinte ("ctb") para pequenas contribuições, autor ("aut") para contribuições maiores. Tradicionalmente, ao citar um pacote em uma publicação científica, apenas os autores "aut" são listados, e não os contribuidores "ctb"; e em `pkgdown` e, nos sites, somente os nomes "aut" são listados na página inicial, e todos os autores são listados na página de autores.

Reviewer: ```suggestion
Se alguém contribuir para o seu repositório, considere adicioná-la em `DESCRIPTION`, como contribuinte ("ctb") para pequenas contribuições, e autor ("aut") para contribuições maiores. Tradicionalmente, ao citar um pacote em uma publicação científica, apenas as pessoas autoras "aut" são listados, e não as contribuidores "ctb"; e em `pkgdown` e, nos sites, somente os nomes "aut" são listados na página inicial, e todas as pessoas autoras são listados na página de autores.
```

> No mínimo, considere adicionar o nome dos colaboradores próximo à linha de correção de *feature*/*bugs* em [NEWS.md](#news).

Reviewer: ```suggestion
No mínimo, considere adicionar o nome das pessoas colaboradoras próximo à linha de correção de *feature*/*bugs* em [NEWS.md](#news).
```

> Recomendamos que você seja generoso com esses agradecimentos, porque é uma coisa boa de se fazer e porque isso aumentará a probabilidade de as pessoas contribuírem novamente com o seu pacote ou com outros repositórios da organização.

Reviewer: ```suggestion
Recomendamos que você seja generoso(a) com esses agradecimentos, porque é uma coisa boa de se fazer e porque isso aumentará a probabilidade de as pessoas contribuírem novamente com o seu pacote ou com outros repositórios da organização.
```

> Como um lembrete de [nossas diretrizes de empacotamento](#building) se o seu pacote foi revisado e você acha que os revisores fizeram uma contribuição substancial para o desenvolvimento do seu pacote, você pode listá-los na seção `Authors@R` com um tipo de contribuinte "Revisor" (`"rev"`), da seguinte forma:

Reviewer: ```suggestion
Como um lembrete de [nossas diretrizes de empacotamento](#building) se o seu pacote foi revisado e você acha que as pessoas revisoras fizeram uma contribuição substancial para o desenvolvimento do seu pacote, você pode listá-las na seção `Authors@R` com um tipo de contribuinte "Revisor" (`"rev"`), da seguinte forma:
```

diretrizes de empacotamento é esquisito, mas precisa ver como está o nome do capítulo citado

Author: Na minha visão, "diretrizes de empacotamento" está ok.

Vários dos termos da programação costumam ficar esquisitos quando traduzidos para o portugues. Nesses casos, é normal evitarmos traduzir o termo para não introduzirmos essa "esquisitisse" dentro do texto. Contudo.... eu não creio que essa seja a solução para esse caso específico de "empacotamento".

Em outras palavras, eu creio que "diretrizes de packaging" ficaria mais esquisito ainda. Então, "diretrizes de empacotamento" cumpre bem o seu papel na minha visão.

```
    person("Bea", "Hernández", role = "rev",
    comment = "Bea reviewed the package (v. X.X.XX) for rOpenSci, see <https://github.com/ropensci/software-review/issues/116>"),
```

> Inclua os revisores somente após solicitar o consentimento deles. Leia mais em [esta postagem do blog "Agradecendo seus revisores: Gratidão por meio de metadados semânticos"](https://ropensci.org/blog/2018/03/16/thanking-reviewers-in-metadata/). Observe que 'rev' gerará uma CRAN NOTE, a menos que o pacote seja criado usando o R v3.5. Certifique-se de que você use `roxygen2` na versão mais recente disponível no CRAN.

Reviewer: ```suggestion
Inclua as pessoas revisoras somente após solicitar o consentimento delas. Leia mais [nesta postagem do blog "Agradecendo seus revisores: Gratidão por meio de metadados semânticos"](https://ropensci.org/blog/2018/03/16/thanking-reviewers-in-metadata/). Observe que 'rev' gerará uma CRAN NOTE, a menos que o pacote seja criado usando o R v3.5. Certifique-se de que você use `roxygen2` na versão mais recente disponível no CRAN.
```

Por favor, não liste os editores como colaboradores. Sua participação e contribuição para a rOpenSci são agradecimentos suficientes!

### Dando as boas-vindas aos colaboradores da rOpenSci {#welcoming-collaborators-to-ropensci}

Se você conceder a alguém permissões de escrita no repositório,

- entre em contato com um [membro da equipe](https://ropensci.org/about#team) para que esse novo colaborador possa receber **um convite para a organização da rOpenSci no GitHub** (em vez de ser [um colaborador externo](https://help.github.com/articles/repository-permission-levels-for-an-organization/#outside-collaborators))

> - entre em contato com a equipe da rOpenSci [ou um outro membro da equipe](https://ropensci.org/about#team) para que esse novo colaborador possa receber **um convite para o espaço de trabalho do Slack da rOpenSci**.

Reviewer: ```suggestion
- entre em contato com a equipe da rOpenSci [ou um outro membro da equipe](https://ropensci.org/about#team) para que esse novo colaborador possa receber **um convite para o *workspace* do Slack da rOpenSci**.
```

Author: Eu acho que "espaço de trabalho do Slack" é uma tradução fiel e clara do termo "Slack workspace". Você realmente vê  alguma vantagem em optarmos por "workspace" ? Essa vantagem não ficou muito clara para mim.

Reviewer: Eu nunca ouvi ninguém falar em "Espaço de trabalho" do Slack... Mesmo na documentação eles chamam de workspace!
https://slack.com/intl/pt-br/help/categories/360000049043
<img width="944" alt="imagem" src="https://github.com/ropensci/dev_guide/assets/42153618/fcb2bf8f-4f19-4b5f-8097-0a975844275b">

Author: Entendido, valeu pela referência @beatrizmilz !

## Outros recursos {#further-resources}

> - Chamada da comunidade rOpenSci [Configure seu pacote para promover uma comunidade](https://ropensci.org/commcalls/apr2021-pkg-community/).

Reviewer: ```suggestion
- Chamada da comunidade rOpenSci: [Configure seu pacote para promover uma comunidade](https://ropensci.org/commcalls/apr2021-pkg-community/).
```
